1_Record_dimension.R

```diff
@@ -1,61 +1,92 @@
+col <- read_sf('/Users/elkintenorio/Library/CloudStorage/GoogleDrive-etenorio@humboldt.org.co/Mi\ unidad/Proyectos/Mapa_de_Vacios/FPV_2024/AMAZONAS', 'AMAZONAS')
+
+# Definición del sistema de referencia geográfico (GCS) y proyectado (CRS)
 GRS.geo<-"+proj=longlat +datum=WGS84 +no_defs +ellps=WGS84 +towgs84=0,0,0" # Geographic Reference to AOI
 CRS.proj<-'+proj=tmerc +lat_0=4.596200416666666 +lon_0=-74.07750791666666 +k=1 +x_0=1000000 +y_0=1000000 +ellps=GRS80 +towgs84=0,0,0,0,0,0,0 +units=m +no_defs' # Planar Reference to AOI
-proj.col<- spTransform(col, CRS.proj)
-shape_zoneOwin<-as.owin(proj.col)
-
-load('data_to_gap.RData') # Records to use in the analysis / require name of the species, coordinates (geograhpic type) and ID identifier
-Data <- data_to_gap
-Data2 <- Data[, c('ID', 'lat', 'lon')]
-Data2 <- Data2[!is.na(Data2$lat),]
-Data2 <- Data2[!is.na(Data2$lon),]
-coordinates(Data2)=~)on+lat
-
-coords<-as.data.frame(unique(Data2@coords))
-coords$ID<-1:nrow(coords)
-coordinates(coords)=~lon+lat
-coords@proj4string@projargs<-GRS.geo
-coordinates.col<- spTransform(coords, CRS.proj)
-system.time(over.coords <- over(coordinates.col, proj.col)); head(over.coords) #Identify coordinates within the AOI
-
-enAOI <- as.numeric(over.coords$OBJECTID)
-enAOI[enAOI>=1] <- 1
-enAOI[is.na(enAOI)] <- 0
-coordinates.col<-as.data.frame(coordinates.col)
-coordinates.col$enAOI <- enAOI
-coordinates.col<- coordinates.col[coordinates.col$enAOI == 1,]
-
-
-#shape_zoneOwin <- as.owin(col)
-p <- ppp(coordinates.col$lon, coordinates.col$lat, window=shape_zoneOwin, unitname=c("metre","metres")) # Create a Point Pattern using unit to GRS
+
+# Transformación del shapefile al sistema de coordenadas proyectadas
+proj.col <- sf::st_transform(col, crs = CRS.proj) 
+
+# Convertir el shapefile a una ventana espacial compatible con spatstat
+shape_zoneOwin <- as.owin(proj.col)
+
+# Cargar registros de especies desde un archivo txt
+Data <- vroom("/Users/elkintenorio/Library/CloudStorage/GoogleDrive-etenorio@humboldt.org.co/Mi\ unidad/Proyectos/Mapa_de_Vacios/FPV_2024/Registros_Agosto_2024/dwc_AMAZONAS.txt", col_names = TRUE)
+Data2 <- Data[, c('gbifID', 'decimalLatitude_x', 'decimalLongitude_x')] #Select ad first column the name of the id record. And for second and third column, latitude and longitude, respectively.
+Data2 <- Data2[!is.na(Data2$decimalLatitude_x),] # Eliminar registros con latitud NA
+Data2 <- Data2[!is.na(Data2$decimalLongitude_x),] # Eliminar registros con longitud NA
+# No estoy seguro que lo que aquí se hace sea correcto. Se está eliminando los duplicados por coordenada
+# Esto implica que se deja un punto por lcalidad, removiendo todas las especies, y dejando solo una
+# Esto puede ser correcto para el segundo código (dimensión ambiental), pero para este (dimensión de registros), no estoy seguro.  
+
+
+# Eliminar duplicados basados en coordenadas
+Data2 <- Data2[!duplicated(Data2[c("decimalLongitude_x", "decimalLatitude_x")]), ]
```

Comment:
This code line isn't part of the script "1_Record_dimention.R" and can't be used in this dimension because it removes essential records to use in this dimension.

Reply:
Hola Cristian, ese comando viene del siguiente comando en el código original:

coords <- as.data.frame(unique(Data2@coords))

Remueve coordenadas duplicadas con una función de paquetes más recientes. Si esa linea es incorrecta en este código, entonces esto viene del código original.